facebook/delphiFacebook/R/contingency_variables.R

```diff
@@ -406,7 +504,402 @@ create_derivative_columns <- function(df) {
+    dontneed_reasons <- split_options(df$V6)
+    
+    df$b_hesitant_dontneed_reason_had_covid <- as.numeric(dontneed & is_selected(dontneed_reasons, "1"))
+    df$b_hesitant_dontneed_reason_dont_spend_time <- as.numeric(dontneed & is_selected(dontneed_reasons, "2"))
+    df$b_hesitant_dontneed_reason_not_high_risk <- as.numeric(dontneed & is_selected(dontneed_reasons, "3"))
+    df$b_hesitant_dontneed_reason_precautions <- as.numeric(dontneed & is_selected(dontneed_reasons, "4"))
+    df$b_hesitant_dontneed_reason_not_serious <- as.numeric(dontneed & is_selected(dontneed_reasons, "5"))
+    df$b_hesitant_dontneed_reason_not_beneficial <- as.numeric(dontneed & is_selected(dontneed_reasons, "7"))
+    df$b_hesitant_dontneed_reason_other <- as.numeric(dontneed & is_selected(dontneed_reasons, "8"))
+  } else {
+    df$b_hesitant_dontneed_reason_had_covid <- NA_real_
+    df$b_hesitant_dontneed_reason_dont_spend_time <- NA_real_
+    df$b_hesitant_dontneed_reason_not_high_risk <- NA_real_
+    df$b_hesitant_dontneed_reason_precautions <- NA_real_
+    df$b_hesitant_dontneed_reason_not_serious <- NA_real_
+    df$b_hesitant_dontneed_reason_not_beneficial <- NA_real_
+    df$b_hesitant_dontneed_reason_other <- NA_real_
+  }
+  
+  # C14
+  if ("b_public_mask_often" %in% names(df)) {
+    df$b_wearing_mask <- as.numeric(df$b_public_mask_often == 1)
+  } else {
+    df$b_wearing_mask <- NA_real_
+  }
+  
+  # B2
+  if ("ms_symptoms" %in% names(df)) {
+    symptoms <- split_options(df$ms_symptoms)
+    
+    ind <- data.frame(dummy = rep(0, nrow(df)))
+    ind$fever <- is_selected(symptoms, "1")
+    ind$cough <- is_selected(symptoms, "2")
+    ind$shortness_of_breath <- is_selected(symptoms, "3")
+    ind$difficulty_breathing <- is_selected(symptoms, "4")
+    ind$tiredness <- is_selected(symptoms, "5")
+    ind$nasal_congestion <- is_selected(symptoms, "6")
+    ind$runny_nose <- is_selected(symptoms, "7")
+    ind$muscle_aches <- is_selected(symptoms, "8")
+    ind$sore_throat <- is_selected(symptoms, "9")
+    ind$chest_pain <- is_selected(symptoms, "10")
+    ind$nausea <- is_selected(symptoms, "11")
+    ind$diarrhea <- is_selected(symptoms, "12")
+    ind$loss_of_smell_or_taste <- is_selected(symptoms, "13")
+    ind$other <- is_selected(symptoms, "14")
+    ind$none <- is_selected(symptoms, "15")
+    ind$eye_pain <- is_selected(symptoms, "16")
+    ind$chills <- is_selected(symptoms, "17")
+    ind$headaches <- is_selected(symptoms, "18")
+    ind$changes_in_sleep <- is_selected(symptoms, "19")
+    ind$missing <- is_selected(symptoms, NA)
+    
+    # COVID-like illness: fever, along with cough, or shortness of breath, or difficulty breathing
+    df$b_cli <- as.numeric(
+      ind$fever & (ind$cough | ind$shortness_of_breath | ind$difficulty_breathing)
+    )
+    
+    # influenza-like illness: fever, along with cough or sore throat
+    df$b_ili <- as.numeric(
+      ind$fever & (ind$cough | ind$sore_throat)
+    )
```

Comment:
We don't need to define the cli/ili variables; the relevant ones are already created in the main pipeline, in the `load_responses` bit. They actually use the household symptoms question (A1) to increase the effective sample size and get a better estimate of CLI/ILI prevalence. The aggregation setup for these items should look like this:

```
    "pct_cli", "n_hh_prop_cli", c("nation"), compute_numeric, jeffreys_count,
    "pct_ili", "n_hh_prop_ili", c("nation"), compute_numeric, jeffreys_count,
```

Reply:
Done.

Reply:
Update following conversation with Sarah: The original definition you used here (ILI/CLI based on individual symptom responses) is apparently correct for the international survey, which doesn't ask the household symptoms question.